lib/m/parser.rb

```diff
@@ -67,8 +67,21 @@ def parse_options!(argv)
           testable.line = line
         end
 
+        opts.on '-r', 'Search provided directory recursively.' do |line|
```

Comment:
I think we can change this to support the option we need it to:

```ruby
opts.on '-r', '--recursive DIR', 'Search provided directory recursively.' do |directory|
  testable.recursive = true
  argv << directory
end
```

This will throw `OptionParser::MissingArgument`, as expected, when running `m -r` without any argument.

What do you think?